### PR TITLE
Remove MatrixType from MGTransferGlobalCoarsening

### PR DESCRIPTION
**Pull request by peterrum**
This simplifies the usage quite a bit.

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 1]
```diff
@@ -247,32 +247,25 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
...
    * Value type.
    */
-  using Number = typename MatrixType::value_type;
+  using Number = typename VectorType::value_type;
 
   /**
-   * Constructor taking an operator for each level (minimum requirement is
-   * that the operator provides the function initialize_dof_vector()) and
-   * transfer operators (with the coarsest level kept empty in @p transfer).
+   * Constructor taking a transfer operators (with the coarsest level kept
```
Comment (marcfehling) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:260]:
```suggestion
   * Constructor taking a transfer operator (with the coarsest level kept
```

**Review by marcfehling — APPROVED** [review]

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.h]: Changed :)

**Review by kronbichler — APPROVED** [review]
This makes sense, yes.

**Comment by peterrum** [comment]
I am merging, since the failing test is unrelated.

**Pull request merged**